ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java

```diff
@@ -101,6 +102,31 @@ public void updateExtraData(ReactTextView view, Object extraData) {
       ReactAccessibilityDelegate.resetDelegate(
           view, view.isFocusable(), view.getImportantForAccessibility());
     }
+
+    CustomLineHeightSpan[] customLineHeightSpans =
+        spannable.getSpans(0, spannable.length(), CustomLineHeightSpan.class);
+    CustomStyleSpan[] customStyleSpans =
+        spannable.getSpans(0, spannable.length(), CustomStyleSpan.class);
+    if (customLineHeightSpans.length > 0 && customStyleSpans.length > 0) {
+      int highestLineHeight = 0;
+      for (CustomLineHeightSpan span : customLineHeightSpans) {
+        if (highestLineHeight == 0 || span.getLineHeight() > highestLineHeight) {
+          highestLineHeight = span.getLineHeight();
```

Comment:
1) Retrieves the line height from the CustomLineHeightSpans (ParagraphSpan)
2) Changes the textPaint baseline based on fontSize and lineHeight of the Paragraph and aligns the text in CustomStyleSpan (MetricAffectingSpan)